pippy/PipelineSchedule.py

```diff
@@ -98,27 +81,34 @@ def __init__(
         num_stages: int,
         rank: int,
         world_size: int,
-        input_args: List[torch.tensor],
-        device: torch.device,
+        inputs_meta: List[torch.tensor],
+        outputs_meta: Optional[List[torch.tensor]] = None,
     ):
         super().__init__()
+        self.module = module
         self.rank = rank
         self.stage_id = stage_id
         self.is_first_stage = stage_id == 0
         self.is_last_stage = stage_id == num_stages - 1
         self.num_stages = num_stages
-        # When we materialize the model partition on cuda, we call reset_parameters() if it is available
```

Comment:
i'm confused about the state of the code before your PR- is the expectation that inputs are real inputs, and model comes on meta device, hence we used to move it to 'real device' and then run real inputs through it?

it must have changed back and forth since the old code `meta_output = self.module(*input_args)` looks half way between each world.  it claims to give meta outputs, but it uses module that's moved to 'device'.

i think in your new PR you are trying to solidify that we use meta inputs and meta model for the PP stage __init__ lifetime, and then 'real' happens sometime later.  If so i agree with that move.